libbeat/publisher/pipeline/processor.go

```diff
@@ -39,7 +39,9 @@ type processorFn struct {
 //  8. (P) pipeline processors list
 //  9. (P) (if publish/debug enabled) log event
 // 10. (P) (if output disabled) dropEvent
-func (p *Pipeline) newProcessorPipeline(
+func newProcessorPipeline(
+	info beat.Info,
```

Comment:
Why did you decide to pass info instead of using it as variable as before?

Reply:
The info is used for the debug print. It used to be part of the pipeline variable. I removed the method and created a constructor passing only those information required to build the processor pipeline. The decoupling simplified the construction of the processor pipeline in the unit tests.